crypto/src/main/scala/coop/rchain/crypto/signatures/Secp256k1.scala

```diff
@@ -37,7 +38,9 @@ object Secp256k1 {
     kpg.initialize(new ECGenParameterSpec(curveName), secureRandomInstance)
     val kp = kpg.generateKeyPair
 
-    val sec = Base16.decode(kp.getPrivate.asInstanceOf[ECPrivateKey].getS().toString(16))
+    val base16Sec = toHex(kp.getPrivate.asInstanceOf[ECPrivateKey].getS.toByteArray)
```

Comment:
No, this is not going to work. `toByteArray` returns a representation of the number which we do not support, see the [scaladocs](https://www.scala-lang.org/api/current/scala/math/BigInt.html#toByteArray:Array[Byte]). We need to give the number's literal value in base 16 and simply add a leading zero if the number of hex-digits is odd.